src/game_api/script/usertypes/socket_lua.cpp

```diff
@@ -94,5 +196,29 @@ void register_usertypes(sol::state& lua)
             DEBUG("Failed hooking network: {}\n", error);
         }
     };
+
+    /// Send a synchronous HTTP GET request and return response as a string or nil on an error
+    lua["http_get"] = [&lua](std::string url) -> sol::optional<std::string>
+    {
+        std::string out;
+        std::string err;
+        if (http_get(url.c_str(), out, err))
+        {
+            return out;
+        }
+        else
+        {
+            luaL_error(lua, err.c_str());
+        }
+        return std::nullopt;
+    };
+
+    /// Send an asynchronous HTTP GET request and run the callback when done. If there is an error, response will be nil and vice versa.
+    /// The callback signature is nil on_data(string response, string error)
+    lua["http_get_async"] = [](std::string url, sol::function on_data) -> HttpRequest*
+    {
+        HttpRequest* req = new HttpRequest(std::move(url), std::move(on_data));
```

Comment:
This is gonna cause race conditions on Lua, isn't it?

Reply:
Don't know, that's why I have tried to keep away from thread stuff.

Reply:
This is the sort of thing that I'm going to need an example and a solution for or I'll just remove the thing cause you said it won't work.

Reply:
Anything that uses `get_calling_backend` and family has a handle to a LuaBackend that is protected by a mutex. Any access to the Lua vm through any other handle is not threadsafe. I don't know the exact names of stuff off the top of my head, but you should be able to compare to anything else that uses callbacks. So the approach would be to `get_calling_backen_id` here, then pass a lambda capturing said id and the cb, inside of which you `get_backend_from_id` or smth and then `handle_function_woth_return<void>` or smth like that. Then you can also store a std:: function in your HttpRequest`. Ig that whole process could be wrapped easily, I'll take a look at it next time I sit at a PC.

Reply:
Thanks. Yeah I kinda just copied the (probably also broken) udp function I made ages ago and it slipped though the radar then. I'll make another branch for fixing the net stuff and merge options in the meantime. I just did http on a whim cause someone mentioned it.